### PR TITLE
[SPARK-30674][INFRA] Use python3 in dev/lint-python

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to use `python3` instead of `python` in `dev/lint-python`.

### Why are the changes needed?

Currently, `dev/lint-python` fails at Python 2. And, Python 2 is EOL since January 1st 2020.
```
$ python -V
Python 2.7.17

$ dev/lint-python
starting python compilation test...
Python compilation failed with the following errors:
Compiling ./python/setup.py ...
  File "./python/setup.py", line 27
    file=sys.stderr)
        ^
SyntaxError: invalid syntax
```

### Does this PR introduce any user-facing change?

No. This is a dev environment.

### How was this patch tested?

Jenkins is running this with Python 3 already.
The following is a manual test.

```
$ python -V
Python 3.8.0

$ dev/lint-python
starting python compilation test...
python compilation succeeded.
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @HyukjinKwon . How do you think about this?

**Comment by dongjoon-hyun** [comment]
Also, cc @viirya . Could you review this PR?

**Comment by viirya** [comment]
hmm, so we currently fail to run `dev/lint-python` in Jenkins?

**Comment by dongjoon-hyun** [comment]
No~ Jenkins is using `Python3`. So, it's okay for now. This is for the other environment.

**Review by viirya — APPROVED** [review]

File: dev/lint-python [hunk 1]
```diff
@@ -38,7 +38,7 @@ function compile_python_test {
 
     # compileall: https://docs.python.org/2/library/compileall.html
```
Comment (viirya) [inline dev/lint-python]: Should we change this comment? Looks like it points to Python 2?

Comment (dongjoon-hyun) [reply dev/lint-python]: Sure. Thanks!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @viirya .

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**